library/pk.c

```diff
@@ -518,6 +518,46 @@ int mbedtls_pk_sign( mbedtls_pk_context *ctx, mbedtls_md_type_t md_alg,
                                          f_rng, p_rng, NULL ) );
 }
 
+#if defined(MBEDTLS_PSA_CRYPTO_C)
+/*
+ * Make a signature with options
+ */
+int mbedtls_pk_sign_ext( mbedtls_pk_type_t pk_type,
+                         mbedtls_pk_context *ctx,
+                         mbedtls_md_type_t md_alg,
+                         const unsigned char *hash, size_t hash_len,
+                         unsigned char *sig, size_t sig_size, size_t *sig_len,
+                         int (*f_rng)(void *, unsigned char *, size_t),
+                         void *p_rng )
+{
+
+    *sig_len = 0;
+
+    if( ctx->pk_info == NULL )
+        return( MBEDTLS_ERR_PK_BAD_INPUT_DATA );
+
+    if( ! mbedtls_pk_can_do( ctx, pk_type ) )
+        return( MBEDTLS_ERR_PK_TYPE_MISMATCH );
+
+    if( pk_type != MBEDTLS_PK_RSASSA_PSS )
+    {
+        return( mbedtls_pk_sign( ctx, md_alg, hash, hash_len,
+                                 sig, sig_size, sig_len, f_rng, p_rng ) );
+    }
+
+#if defined(MBEDTLS_X509_RSASSA_PSS_SUPPORT) && defined(MBEDTLS_USE_PSA_CRYPTO)
```

Comment:
I think this should be just `#if defined(MBEDTLS_X509_RSASSA_PSS_SUPPORT)`. The whole function already depends on `PSA_CRYPTO_C` so we can just use PSA Crypto already without checking any further macro.

This is different from `pk_verify_ext()`, for several reasons, one of them being that `pk_verify_ext()` existed before PSA Crypto - here it's a new function so we can just say "this function is only available if `PSA_CRYPTO_C` is enabled and it always uses PSA Crypto, period". (Which is what we're doing for TLS 1.3: depends on `PSA_CRYPTO_C` and always uses PSA Crypto - also because it's new and not encumbered by compatibility issues.)

Reply:
Note: I was wrong, things in PK (more generally, libmedcrypto) should not depend on X.509 config options. X.509 depends on PK, not the other way round, let's avoid reverse dependencies. Fortunately you fixed that in the meantime, so no further change needed - I just wanted to correct my previous comment.